### PR TITLE
CosmosDB autopilot support

### PR DESCRIPTION
**Pull request by ravgill**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by markcowl** [comment]
@ravgill  This spec needs a .Net readme file.  Please add this.

**Review by markcowl — CHANGES_REQUESTED** [review]
Failing validation because there is no .net readme file for this spec - please add a .Net readme file.

**Comment by ravgill** [comment]
@markcowl can you please re-review this?

**Review by MehaKaushik — APPROVED** [review]

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-12-12/cosmos-db.json [hunk 1]
```diff
@@ -5885,11 +5889,54 @@
...
+        },
+        "targetMaxThroughput": {
+          "type": "integer",
+          "description": "Represents target max throughput an autopilot container should operate at once offer is no longer in pending state.",
+          "readOnly": true
+        }
       },
       "required": [
-        "throughput"
+        "maxThroughput"
```
Comment (markcowl) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-12-12/cosmos-db.json]: It looks like 'throughput is now not required for 'ThroughpoutSettingsResource', is that intentional, is this no longer a required property?

**Review by markcowl — CHANGES_REQUESTED** [review]

**Comment by markcowl** [comment]
@ravgill  One question, otherwise lgtm

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-12-12/cosmos-db.json (hunk 1 above)
Comment (ravgill) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-12-12/cosmos-db.json]: Either "throughput" is required or "autopilotSettings" settings is required. Is there a some way to convey "requiredOneOf" @markcowl

Comment (markcowl) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-12-12/cosmos-db.json]: Unfortunately, not until Swagger 3.  It is not a breakign change to remove 'required', but I would suggest documenting the fact that one of these are required in the description of the object and in the description for both parameters

Comment (ravgill) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-12-12/cosmos-db.json]: @markcowl I have updated the description of the object and in the description for both parameters

**Review by markcowl — APPROVED** [review]

**Pull request merged**